apps/callscreen/js/calls_handler.js

```diff
@@ -859,35 +883,46 @@ var CallsHandler = (function callsHandler() {
     }
   }
 
+  /**
+   * Adjusts the state of the hold/merge button to reflect the current calls'
+   * state. If only one call is available the hold button alone will be
+   * displayed if the call's switchable. The state of the button will depend on
+   * the call being on hold or not. If two calls are being handled at the same
+   * time we'll display the merge button if the second call's mergeable. If not
+   * no button will be displayed at all. We don't support cases where more than
+   * two calls are being handled at the same time; this code will need to be
+   * revisited if CALLS_LIMIT is increased above 2.
+   */
   function updateMergeAndOnHoldStatus() {
```

Comment:
This function is not tested thoroughly enough. The changes to the tests only get them passing; they aren't augmented to cover the new functionality.